### PR TITLE
Separated out the modules.d file management into a new resource type …

### PR DESCRIPTION
**Pull request by triforce**
…that allows the defining of modules.d files as erb templates, static content files and preserving the original functionality as well. The new type can also be used as an exported resource which is collected by the metricbeat::config class.

Added a default configuration parameter to allow the use of modules.d files.

### CODE REVIEW

**Comment by corey-hammerton** [comment]
@triforce Please merge from the master branch and test again.

**Comment by triforce** [comment]
> @triforce Please merge from the master branch and test again.

Done.

**Comment by corey-hammerton** [comment]
@triforce The Travis-CI tests are still failing. Please investigate.

**Comment by triforce** [comment]
> @triforce The Travis-CI tests are still failing. Please investigate.

Yep looking into it now.

**Comment by triforce** [comment]
Build fixed.

File: manifests/modulesd.pp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  file { "${metricbeat::config_dir}/modules.d/${template_name}.yml":
+    ensure  => present,
+    source  => $default_source,
+    content => $content,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    require => Class['::metricbeat'],
+    notify  => Service['metricbeat'],
```
[the remark this answers is not in the document]
Comment (triforce) [reply manifests/modulesd.pp]: OK sounds sensible

**Pull request merged**